[CI] RestControllerIT testMetricsEmittedOnSuccess failing

Run over 1000 times locally, but fails on CI quite often. Muting for now, will investigate why.

**Build scan:**
https://gradle-enterprise.elastic.co/s/owrnohwg3vpea/tests/:server:internalClusterTest/org.elasticsearch.rest.RestControllerIT/testMetricsEmittedOnSuccess


**Reproduction line:**
```
./gradlew ':server:internalClusterTest' --tests "org.elasticsearch.rest.RestControllerIT.testMetricsEmittedOnSuccess" -Dtests.seed=4F03A9D49B811263 -Dtests.locale=sq-AL -Dtests.timezone=America/Pangnirtung -Druntime.java=22
```

**Applicable branches:**
main

**Reproduces locally?:**
No

**Failure history:**
[Failure dashboard for `org.elasticsearch.rest.RestControllerIT#testMetricsEmittedOnSuccess`](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('testMetricsEmittedOnSuccess'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.rest.RestControllerIT'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))


**Failure excerpt:**
```
java.lang.AssertionError: 
Expected: a collection with size <1>
     but: collection size was <0>

  at __randomizedtesting.SeedInfo.seed([4F03A9D49B811263:64EF9D2613AF0C03]:0)
  at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
  at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:6)
  at org.elasticsearch.test.ESTestCase.assertThat(ESTestCase.java:2303)
  at org.elasticsearch.rest.RestControllerIT.assertMeasurement(RestControllerIT.java:109)
  at org.elasticsearch.rest.RestControllerIT.testMetricsEmittedOnSuccess(RestControllerIT.java:65)
  at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
  at java.lang.reflect.Method.invoke(Method.java:580)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.junit.rules.RunRules.evaluate(RunRules.java:20)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.junit.rules.RunRules.evaluate(RunRules.java:20)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at org.junit.rules.RunRules.evaluate(RunRules.java:20)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:1570)

```

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -53,8 +53,6 @@ tests:
 - class: org.elasticsearch.index.store.FsDirectoryFactoryTests
   method: testPreload
   issue: https://github.com/elastic/elasticsearch/issues/110211
-- class: "org.elasticsearch.rest.RestControllerIT"
-  issue: "https://github.com/elastic/elasticsearch/issues/110225"
 - class: org.elasticsearch.upgrades.SecurityIndexRolesMetadataMigrationIT
   method: testMetadataMigratedAfterUpgrade
   issue: https://github.com/elastic/elasticsearch/issues/110232

--- a/server/src/internalClusterTest/java/org/elasticsearch/rest/RestControllerIT.java
+++ b/server/src/internalClusterTest/java/org/elasticsearch/rest/RestControllerIT.java
@@ -20,9 +20,12 @@ import org.elasticsearch.common.settings.IndexScopedSettings;
 import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.common.settings.SettingsFilter;
 import org.elasticsearch.features.NodeFeature;
+import org.elasticsearch.logging.LogManager;
+import org.elasticsearch.logging.Logger;
 import org.elasticsearch.plugins.ActionPlugin;
 import org.elasticsearch.plugins.Plugin;
 import org.elasticsearch.plugins.PluginsService;
+import org.elasticsearch.plugins.TelemetryPlugin;
 import org.elasticsearch.telemetry.Measurement;
 import org.elasticsearch.telemetry.TestTelemetryPlugin;
 import org.elasticsearch.test.ESIntegTestCase;
@@ -34,10 +37,10 @@ import java.util.List;
 import java.util.function.Consumer;
 import java.util.function.Predicate;
 import java.util.function.Supplier;
-import java.util.stream.Stream;
 
 import static org.hamcrest.Matchers.hasEntry;
 import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.instanceOf;
 import static org.hamcrest.Matchers.is;
 
 @ESIntegTestCase.ClusterScope(scope = ESIntegTestCase.Scope.TEST, numClientNodes = 1, numDataNodes = 0)
@@ -54,7 +57,7 @@ public class RestControllerIT extends ESIntegTestCase {
         assertEquals(ChunkedResponseWithHeadersPlugin.HEADER_VALUE, response.getHeader(ChunkedResponseWithHeadersPlugin.HEADER_NAME));
     }
 
-    public void testMetricsEmittedOnSuccess() throws IOException {
+    public void testMetricsEmittedOnSuccess() throws Exception {
         final var client = getRestClient();
         final var request = new Request("GET", TestEchoStatusCodePlugin.ROUTE);
         request.addParameter("status_code", "200");
@@ -70,7 +73,7 @@ public class RestControllerIT extends ESIntegTestCase {
         });
     }
 
-    public void testMetricsEmittedOnRestError() throws IOException {
+    public void testMetricsEmittedOnRestError() throws Exception {
         final var client = getRestClient();
         final var request = new Request("GET", TestEchoStatusCodePlugin.ROUTE);
         request.addParameter("status_code", "503");
@@ -85,7 +88,7 @@ public class RestControllerIT extends ESIntegTestCase {
         });
     }
 
-    public void testMetricsEmittedOnWrongMethod() throws IOException {
+    public void testMetricsEmittedOnWrongMethod() throws Exception {
         final var client = getRestClient();
         final var request = new Request("DELETE", TestEchoStatusCodePlugin.ROUTE);
         final var response = expectThrows(ResponseException.class, () -> client.performRequest(request));
@@ -97,30 +100,38 @@ public class RestControllerIT extends ESIntegTestCase {
         });
     }
 
-    private static void assertMeasurement(Consumer<Measurement> measurementConsumer) {
-        var measurements = new ArrayList<Measurement>();
-        for (PluginsService pluginsService : internalCluster().getInstances(PluginsService.class)) {
-            final TestTelemetryPlugin telemetryPlugin = pluginsService.filterPlugins(TestTelemetryPlugin.class).findFirst().orElseThrow();
-            telemetryPlugin.collect();
+    private void assertMeasurement(Consumer<Measurement> measurementConsumer) throws Exception {
+        assertBusy(() -> {
+            var measurements = new ArrayList<Measurement>();
+            for (var nodeName : internalCluster().getNodeNames()) {
+                PluginsService pluginsService = internalCluster().getInstance(PluginsService.class, nodeName);
+                var telemetryPlugins = pluginsService.filterPlugins(TelemetryPlugin.class).toList();
 
-            final var metrics = telemetryPlugin.getLongCounterMeasurement(RestController.METRIC_REQUESTS_TOTAL);
-            measurements.addAll(metrics);
-        }
-        assertThat(measurements, hasSize(1));
-        measurementConsumer.accept(measurements.get(0));
+                assertThat(telemetryPlugins, hasSize(1));
+                assertThat(telemetryPlugins.get(0), instanceOf(TestTelemetryPlugin.class));
+                var telemetryPlugin = (TestTelemetryPlugin) telemetryPlugins.get(0);
+
+                telemetryPlugin.collect();
+
+                final var metrics = telemetryPlugin.getLongCounterMeasurement(RestController.METRIC_REQUESTS_TOTAL);
+                logger.info("collecting [{}] metrics from [{}]", metrics.size(), nodeName);
+                measurements.addAll(metrics);
+            }
+            assertThat(measurements, hasSize(1));
+            measurementConsumer.accept(measurements.get(0));
+        });
     }
 
     @Override
     protected Collection<Class<? extends Plugin>> nodePlugins() {
-        return Stream.concat(
-            super.nodePlugins().stream(),
-            Stream.of(ChunkedResponseWithHeadersPlugin.class, TestEchoStatusCodePlugin.class, TestTelemetryPlugin.class)
-        ).toList();
+        return List.of(ChunkedResponseWithHeadersPlugin.class, TestEchoStatusCodePlugin.class, TestTelemetryPlugin.class);
     }
 
     public static class TestEchoStatusCodePlugin extends Plugin implements ActionPlugin {
         static final String ROUTE = "/_test/echo_status_code";
         static final String NAME = "test_echo_status_code";
+
+        private static final Logger logger = LogManager.getLogger(TestEchoStatusCodePlugin.class);
 
         @Override
         public Collection<RestHandler> getRestHandlers(
@@ -148,7 +159,8 @@ public class RestControllerIT extends ESIntegTestCase {
                 @Override
                 protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient client) {
                     var statusCode = request.param("status_code");
-                    client.getLocalNodeId();
+                    logger.info("received echo request for {}", statusCode);
+
                     var restStatus = RestStatus.fromCode(Integer.parseInt(statusCode));
                     return channel -> {
                         final var response = RestResponse.chunked(
@@ -161,6 +173,7 @@ public class RestControllerIT extends ESIntegTestCase {
                             null
                         );
                         channel.sendResponse(response);
+                        logger.info("sent response");
                     };
                 }
             });

